storage/core/src/common/Interfaces.ts

```diff
@@ -52,3 +52,53 @@ export interface ConfigTransferInput {
   reference: ObjectReference;
   transferConfig: TransferConfig;
 }
+
+export interface DirectoryTransferConfigInput {
+  baseDirectory: BaseDirectory;
+  transferConfig: TransferConfig;
+}
+
+export class EntityPageListIterator<TEntity>
```

Comment:
It is better not export implementations where we do not need to. We return an interface from the functions, and that is sufficient to export, the implementation can remain internal, will be easier to change it if needed.